src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Steps/GetApprenticeshipSteps.cs

```diff
@@ -19,60 +16,67 @@ public class GetApprenticeshipSteps
         private readonly TestContext _context;
         private Fixture _fixture = new Fixture();
         private Apprentice _apprentice;
+        private Apprenticeship _apprenticeship;
 
         public GetApprenticeshipSteps(TestContext context)
         {
             _context = context;
-
-            _apprentice = _fixture.Build<Apprentice>()
-                .Create();
-            _apprentice.AddApprenticeship(_fixture.Create<Apprenticeship>());
+            _apprentice = _fixture.Build<Apprentice>().Create();
+            _apprenticeship = _fixture.Create<Apprenticeship>();
         }
 
-        [Given("there is one apprenticeship")]
-        public async Task GivenThereIsOneApprenticeship()
+        [Given(@"the apprenticeship exists and it's associated with this apprentice")]
+        public async Task GivenTheApprenticeshipExistsAndItSAssociatedWithThisApprentice()
         {
+            _apprentice.AddApprenticeship(_apprenticeship);
             _context.DbContext.Apprentices.Add(_apprentice);
             await _context.DbContext.SaveChangesAsync();
         }
 
-        [Given("there are no apprenticeships")]
-        public async Task GivenThereAreNoApprenticeships()
+        [Given(@"there is no apprenticeship")]
+        public void GivenThereIsNoApprenticeship()
+        {
+        }
+
+        [Given(@"the apprenticeship exists, but it's associated with another apprentice")]
+        public async Task GivenTheApprenticeshipExistsButItSAssociatedWithAnotherApprentice()
         {
+            var anotherApprentice = _fixture.Create<Apprentice>();
+            anotherApprentice.AddApprenticeship(_apprenticeship);
+
+            _context.DbContext.Apprentices.Add(anotherApprentice);
+            _context.DbContext.Apprentices.Add(_apprentice);
+            await _context.DbContext.SaveChangesAsync();
         }
 
-        [When("we try to retrieve the apprenticeships")]
-        public async Task WhenWeTryToRetrieveTheApprenticeships()
+        [When(@"we try to retrieve the apprenticeship")]
+        public async Task WhenWeTryToRetrieveTheApprenticeship()
         {
-            await _context.Api.Get($"apprentices/{_apprentice.Id}/apprenticeships");
+            await _context.Api.Get($"apprentices/{_apprentice.Id}/apprenticeships/{_apprenticeship.Id}");
         }
 
-        [Then("the result should return ok")]
+        [Then(@"the result should return ok")]
```

Comment:
I wonder if we should have a `GeneralSteps` class that contains this?  We seem to reproduce it in every steps file!

Reply:
Good idea!